pyomo/core/expr/numeric_expr.py

```diff
@@ -1376,7 +1376,7 @@ def create_node_with_local_data(self, args, classtype=None):
                 classtype = self.__class__
             elif ( args[0].__class__ is MonomialTermExpression or
                    (args[0].__class__ in native_types or args[0].is_constant()
-                ) and all(arg.__class__ is MonomialTermExpression
+                ) and len(args) > 1 and all(arg.__class__ is MonomialTermExpression
                           for arg in args[1:])):
                 classtype = self.__class__
```

Comment:
This proposed change is IMHO to aggressive at turning valid a `LinearExpression` into a general `SumExpression`.  I think a better solution is:
```python
if classtype is None:
    const = 0
    new_args = []
    for arg in args:
        if arg.__class__ is MonomialTermExpression:
            new_args.append(arg)
        elif arg.__class__ in native_types or arg.is_constant():
            const += arg
        else:
            return SumExpression(args)
    return self.__class__([const] + new_args)
```

Reply:
Does it make sense to have a `LinearExpression` with no variables?

Reply:
Probably not.  So, after the loop might look like:
```python
if not new_args:
    return const
if const:
    new_args.insert(0, const)
return self.__class__(new_args)
```

Reply:
@michaelbynum Following up on today's discussion:  what do you think about the following implementation:
```python
def create_node_with_local_data(self, args, classtype=None):
    if classtype is not None:
        return classtype(args)
    else:
        const = 0
        new_args = []
        for arg in args:
            if arg.__class__ is MonomialTermExpression:
                new_args.append(arg)
            elif arg.__class__ in native_types or arg.is_constant():
                const += arg
            else:
                return SumExpression(args)
        if not new_args:
            return const
        if const:
            new_args.insert(0, const)
        return self.__class__(new_args)
```

Reply:
@jsiirola, I like this proposal a lot. I'm on board.